assembl/static2/js/app/pages/idea.jsx

```diff
@@ -145,11 +134,25 @@ class Idea extends React.Component {
         </div>
       );
     }
-
     const { idea } = ideaData;
-
-    const isUserConnected = getConnectedUserId();
-
+    const isMultiColumn = ideaWithPostsData.loading ? undefined : ideaWithPostsData.idea.messageColumns.length > 0;
+    const messageColumns = ideaWithPostsData.loading ? undefined : ideaWithPostsData.idea.messageColumns;
+    const childProps = {
+      idea: idea,
+      ideaWithPostsData: ideaWithPostsData,
+      isUserConnected: getConnectedUserId(),
+      contentLocaleMapping: contentLocaleMapping,
+      refetchIdea: refetchIdea,
+      lang: lang,
+      noRowsRenderer: noRowsRenderer,
+      messageColumns: messageColumns,
+      getTopPosts: this.getTopPosts,
```

Comment:
Why do we need to pass the method and the data downstream? Can't the downstream component call the method instead? Or are we using `posts` array as a prop to trigger changes?

Reply:
good catch, I should just pass posts